tor/cmd_info.go

```diff
@@ -0,0 +1,65 @@
+package tor
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrServiceNotCreated is used when we want to query info on an onion
+	// service while it's not been created yet.
+	ErrServiceNotCreated = errors.New("onion service hasn't been created")
+
+	// ErrServiceNotFound is used when we query a given onion service info
+	// but the Tor service can't find it.
+	ErrServiceNotFound = errors.New("onion service not found")
+)
+
+// GetServiceInfo uses the onion service ID, which is an onion address without
+// the ".onion" part, to query the Tor service about its descriptor info. The
+// main purpose of using the endpoint "hs/service/desc/id/<ADDR>" is to check
+// the liveness of the hidden onion service, thus the descriptor info is
+// discarded. Tor server returns code 250 if succeeded, 551 if the service is
+// unrecognized or not found in its local descriptor cache.
+func (c *Controller) GetServiceInfo() error {
+	// Check that we have a hidden service created.
+	if c.activeServiceID == "" {
+		return ErrServiceNotCreated
+	}
+
+	// Fetch the descriptor info on the service.
+	cmd := fmt.Sprintf("GETINFO hs/service/desc/id/%s", c.activeServiceID)
+	code, reply, err := c.sendCommand(cmd)
+	if err == nil {
+		return nil
+	}
+
+	log.Debugf("query service:%v got err:%v, reply:%v",
+		c.activeServiceID, err, reply)
+
+	switch code {
+
+	// If we have an error, we'd expect the code to NOT be 250. So we log
+	// an error here if that happens.
+	// TODO(yy): unify the usage of err and code so we could rely on a
+	// single source to change our state.
+	case success:
+		log.Errorf("server replied 250, but with an err:%v", err)
+		return nil
```

Comment:
Yet we don't do anything with this? Or this is just mean to allow us to check if the tor daemon knows of the service or not?

Reply:
Ok this usage of the endpoint seems to have created some confusion. I've switched to a different endpoint, the `GETINFO onions/current`, which would also save us a few bandwidths as the rely is much smaller.